api-gateway/src/main/kotlin/com/saveourtool/save/gateway/utils/StoringServerAuthenticationSuccessHandler.kt

```diff
@@ -28,18 +30,21 @@ class StoringServerAuthenticationSuccessHandler(
     ): Mono<Void> {
         logger.info("Authenticated user ${authentication.userName()} with authentication type ${authentication::class}, will send data to backend")
 
-        val user = authentication.toUser().apply {
-            // https://github.com/saveourtool/save-cloud/issues/583
-            // fixme: this sets a default role for a new user with minimal scope, however this way we discard existing role
-            // from authentication provider. In the future we may want to use this information and have a mapping of existing
-            // roles to save-cloud roles.
-            role = Role.VIEWER.asSpringSecurityRole()
+        val (source, nameInSource) = if (authentication is OAuth2AuthenticationToken) {
```

Comment:
not really understand, do we need source at all or not?

Reply:
I think we don't need at all. I will continue to investigate later. But as I get -- we make a first login by OAUTH2 or Basic Auth on Gateway side. Then further we send only `user` or `user_id` to backend. Backend is in private network and all connections to it are authorised and authenticated.

Reply:
yes, we don't need `source` on backend side. We need it only in api-gateway to resolve save's user name